### PR TITLE
Chapter 1 Homework 3

### PR DESCRIPTION
**Pull request by calvareztx**
Created a new class called Lesson3 and added new functions including a Demo function called from Program class. Updated Checks.cs to include calls to my functions implementation from Lesson3 class.

### CODE REVIEW

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4863638)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 3
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4863638&bid=16131262)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+using System.Globalization;
+using System.Text;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4863638)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            string userSurname = UserName(message: "Enter your surname: ");
+            int userAge = UserAge(message: "Enter your age: ");
+            float userHeight = UserHeightWeight(message: "Enter your height in cm: ");
+            float userWeight = UserHeightWeight(message: "Enter your weight in kg: ");
+            float userBMI = CalculateBMI(userWeight, userHeight);
+            Console.WriteLine(userFirstName + " " + userSurname + " is " + userAge + " years old, his weight is " + userWeight + " kg and his height is " + userHeight + " cm.");
+            Console.WriteLine("BMI = " + userBMI);
+        }
+
+        public static float CalculateBMI(float weight, float height)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [Rename method 'CalculateBMI' to match pascal case naming rules, consider using 'CalculateBmi'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4863638)

File: Src/BootCamp.Chapter/Lesson3.cs (hunk 1 above)
Comment (calvareztx) [reply Src/BootCamp.Chapter/Lesson3.cs]: Removed.

File: Src/BootCamp.Chapter/Lesson3.cs (hunk 2 above)
Comment (calvareztx) [reply Src/BootCamp.Chapter/Lesson3.cs]: Removed.

File: Src/BootCamp.Chapter/Lesson3.cs (hunk 3 above)
Comment (calvareztx) [reply Src/BootCamp.Chapter/Lesson3.cs]: Updated method name.

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4863638&bid=16131262)

File: Src/BootCamp.Chapter/Checks.cs [hunk 4]
```diff
@@ -17,25 +13,25 @@ public static class Checks
         public static int PromptInt(string message)
         {
             // To do: call your implementation. 
-            return 0;
+            return Lesson3.UserAge(message);
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Checks.cs]: Remove TODO comments when you implemented them. :)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    public class Lesson3
+    {
+        public static void Demo ()
+        {
+            string userFirstName = UserName(message: "Enter your first name: ");
+            string userSurname = UserName(message: "Enter your surname: ");
+            int userAge = UserAge(message: "Enter your age: ");
+            float userHeight = UserHeightWeight(message: "Enter your height in cm: ");
+            float userWeight = UserHeightWeight(message: "Enter your weight in kg: ");
+            float userBMI = CalculateBmi(userWeight, userHeight);
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Lesson3.cs]: Those function names are reflecting what you want to do with them but the functionality differs from that. Imagine you want to reuse UserName() to request the name of his town or UserAge() to ask for the amount of siblings the user has.
This won't work and therefor the choice of names here is bad. Please fix! :)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+            Console.WriteLine(userFirstName + " " + userSurname + " is " + userAge + " years old, his weight is " + userWeight + " kg and his height is " + userHeight + " cm.");
+            Console.WriteLine("BMI = " + userBMI);
+        }
+
+        public static float CalculateBmi(float weight, float height)
+        {
+            double bmi = weight / Math.Pow(height, 2);
+            return Convert.ToSingle(bmi);
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Lesson3.cs]: You're doing quite heavy stuff here which you don't need to do.
Consider using float bmi = weight / (height * height) to spare the unneccessary extra work.

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        }
+
+        public static float UserHeightWeight(string message)
+        {
+            Console.Write(message);
+            var heightOrWeight = Console.ReadLine();
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Lesson3.cs]: If you run into variable names like this your alarm bell should ring. You're probably doing something in a rather unhealthy fashion.

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+            Console.Write(message);
+            var name = Console.ReadLine();
+            return name;
+        }
+
+        public static int UserAge(string message)
+        {
+            Console.Write(message);
+            var age = Console.ReadLine();
+            return Convert.ToInt32(age);
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Lesson3.cs]: Consider using int.Parse() here.

File: Src/BootCamp.Chapter/Program.cs [hunk 9]
```diff
@@ -9,6 +9,12 @@ class Program
...
         {
+            Lesson3.Demo();
         }
     }
 }
+
+
+
+
+
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Program.cs]: Please remove empty lines. :)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+            Console.Write(message);
+            var name = Console.ReadLine();
+            return name;
+        }
+
+        public static int GetIntInput(string message)
+        {
+            Console.Write(message);
+            var age = Console.ReadLine();
+            //return Convert.ToInt32(age);
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4863638)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 1
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4863638&bid=16131262)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4863638&bid=16131262)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 11]
```diff
@@ -7,41 +7,41 @@ public class Lesson3
...
             Console.Write(message);
             var name = Console.ReadLine();
             return name;
         }
 
-        public static int UserAge(string message)
+        public static int GetIntInput(string message)
         {
             Console.Write(message);
             var age = Console.ReadLine();
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Lesson3.cs]: You forgot to change that variable name.

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4863638&bid=16131262)

**Comment by calvareztx** [comment]
Thank you for the approval. Kindly close at your convenience.

**Pull request closed without merging**